### PR TITLE
value arr size input

### PR DESCRIPTION
**Pull request by volodymyr-basiuk**
- added `valueArraySize` input
- disable `SMTVerifier` for NOOP operator
- added new IN template (InWithDynamicArraySize) which works with `valueArraySize` input 
- removed testvectors from repo

### CODE REVIEW

File: circuits/lib/query/comparators.circom [hunk 1]
```diff
@@ -24,6 +24,30 @@ template IN (valueArraySize){
...
 }
 
+// Same as IN but stops checking on valueArraySize
+template InWithDynamicArraySize (maxValueArraySize){
+        signal input in;
+        signal input value[maxValueArraySize];
+        signal input valueArraySize;
+        signal output out;
+
+        assert(maxValueArraySize < 256);
```
Comment (OBrezhniev) [inline circuits/lib/query/comparators.circom:34]: <= 256

File: circuits/lib/utils/arraySizeValidator.circom [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    signal sizeEqOne <== IsEqual()([valueArraySize, 1]);
+    signal sizeEqTwo <== IsEqual()([valueArraySize, 2]);
+    signal sizeLessOrEqMax <== LessThan(8)([valueArraySize, maxValueArraySize + 1]);
+
+    signal sizeNotEqZero <== NOT()(sizeEqZero);
+    signal moreThanZeroLessOrEqMax <== AND()(sizeNotEqZero, sizeLessOrEqMax);
+
+    signal opBits[5] <== Num2Bits(5)(operator); // values 0-15 are query operators, 16-31 - modifiers/computations
+
+    assert(maxValueArraySize < 256);
```
Comment (OBrezhniev) [inline circuits/lib/utils/arraySizeValidator.circom:44]: <= 256

File: circuits/lib/utils/arraySizeValidator.circom [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+    // query operator mux
+    component mux = Mux4();
+    mux.s <== [opBits[0], opBits[1], opBits[2], opBits[3]];
+
+    // We don't use 5th bit (opBits[4]) here; which specifies whether operator is query or
+    // modifier/computation operator. It's used in the final mux.
+    _ <== opBits[4];
```
Comment (OBrezhniev) [inline circuits/lib/utils/arraySizeValidator.circom:52]: We need to verify array size for modifier ops too.

File: testvectorgen/credentials/onchain/v3/v3_test.go [hunk 4]
```diff
@@ -238,7 +239,7 @@ func Test_Nullify(t *testing.T) {
 	desc := "Nullify"
 	isUserIDProfile := true
 	isSubjectIDProfile := true
-	value := utils.PrepareStrArray([]string{"94313"}, 64)
+	value := []string{}
```
Comment (OBrezhniev) [inline testvectorgen/credentials/onchain/v3/v3_test.go:242]: It shouldn't be empty I think.

File: testvectorgen/credentials/v3/v3_test.go [hunk 5]
```diff
@@ -214,16 +216,16 @@ func Test_Nullify(t *testing.T) {
 	desc := "Nullify"
 	isUserIDProfile := true
 	isSubjectIDProfile := true
-	value := utils.PrepareStrArray([]string{"94313"}, 64)
-	generateTestDataWithOperatorAndRevCheck(t, desc, isUserIDProfile, isSubjectIDProfile, "0", "123", "mtp/nullify", utils.NOOP, &value, false, 1, false, Mtp)
-	generateTestDataWithOperatorAndRevCheck(t, desc, isUserIDProfile, isSubjectIDProfile, "0", "123", "sig/nullify", utils.NOOP, &value, false, 1, false, Sig)
+	value := []string{}
```
Comment (OBrezhniev) [inline testvectorgen/credentials/v3/v3_test.go:219]: Same here.

File: testvectorgen/utils/utils.go [hunk 6]
```diff
@@ -316,3 +330,30 @@ func CalculateNullify(genesisID, claimSubjectProfileNonce, claimSchema, verifier
 
 	return nullifier.String(), nil
 }
+
+func GetValueArraySizeForOperator(operator int) int {
+	result := 0
+	oneArrLengthOps := []int{1, 2, 3, 6, 7, 8}
+	twoArrLengthOps := []int{9}
```
Comment (OBrezhniev) [inline testvectorgen/utils/utils.go:337]: , 10

**Review by OBrezhniev — CHANGES_REQUESTED** [review]

File: circuits/lib/query/comparators.circom (hunk 1 above)
Comment (volodymyr-basiuk) [reply circuits/lib/query/comparators.circom]: fixed

File: circuits/lib/utils/arraySizeValidator.circom (hunk 2 above)
Comment (volodymyr-basiuk) [reply circuits/lib/utils/arraySizeValidator.circom]: fixed

File: testvectorgen/credentials/v3/v3_test.go (hunk 5 above)
Comment (volodymyr-basiuk) [reply testvectorgen/credentials/v3/v3_test.go:219]: Also test with NOOP op here

File: testvectorgen/utils/utils.go (hunk 6 above)
Comment (volodymyr-basiuk) [reply testvectorgen/utils/utils.go]: Added

**Review by OBrezhniev — APPROVED** [review]

**Pull request merged**